src/main/webapp/src/components/autocompleteTextField.js

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import { withStyles } from "@material-ui/styles";
+import TextField from '@material-ui/core/TextField';
+import Autocomplete from '@material-ui/lab/Autocomplete';
+import Typography from '@material-ui/core/Typography';
+import throttle from 'lodash/throttle';
+import axios from "axios";
+
+/**
+ * Change TextField outline border default colors.
+ */
+const StyledTextField = withStyles({
+  root: {
+    "& label.Mui-focused": {
+      color: "black",
+    },
+    "& .MuiInput-underline:after": {
+      borderBottomColor: "#733F94",
+    },
+    "& .MuiOutlinedInput-root": {
+      "&.Mui-focused fieldset": {
+        borderColor: "#733F94",
+      },
+    },
+  },
+})(TextField);
+
+export default function AutocompleteTextField(props) {
+  const [value, setValue] = React.useState(props.value);
+  const [options, setOptions] = React.useState([]);
+
+  const fetch = React.useMemo(
+    () =>
+      throttle((request, callback) => {
+        axios
+          .post(`/autocomplete?type=${props.type}&searchString=${request}`)
+          .then((result) => result.data)
+          .then((options) => callback(options));
+      }, 200),
+    [],
+  );
+
+  React.useEffect(() => {
+    let active = true;
+
+    if (value === '') {
+      return undefined;
+    }
+
+    fetch(value, (results) => {
+      if (active) {
```

Comment:
Why do you need the `active` flag at all?

Reply:
Because we have async call in it and we need to control that we change state of smth that still exists. Longer explanation can be found [here](https://stackoverflow.com/questions/62453435/why-do-we-need-to-return-a-function-in-the-react-hook).